lib/ret_web/channels/hub_channel.ex

```diff
@@ -455,6 +456,84 @@ defmodule RetWeb.HubChannel do
     {:noreply, socket}
   end
 
+  def handle_in("list_entity_states", _, socket) do
+    hub = socket |> hub_for_socket
+    entity_states = EntityState.list_entity_states(hub.hub_id)
+    {:reply, {:ok, EntityStateView.render("index.json", %{entity_states: entity_states})}, socket}
+  end
+
+  def handle_in(
+        "create_entity_state",
+        %{"nid" => nid, "create_message" => create_message, "updates" => updates},
+        socket
+      ) do
+    with {:ok, hub} <- authorize(socket, :write_entity_state),
+         {:ok, entity_state} <-
+           EntityState.create_entity_state!(hub, %{
+             nid: nid,
+             create_message: create_message,
+             updates: updates
+           }) do
+      broadcast!(
+        socket,
+        "entity_state_created",
+        EntityStateView.render("show.json", %{entity_state: entity_state})
+      )
+
+      {:reply, :ok, socket}
+    else
+      {:error, %{reason: reason}} ->
+        {:reply, {:error, %{reason: reason}}, socket}
+
+      {:error, other} ->
+        IO.inspect({:error, other})
+        {:reply, {:error, %{reason: "TODO"}}, socket}
```

Comment:
I left this `TODO` because I knew I did not account for the case where `create_entity_state!` fails. I expect to happen whenever two clients attempt to pin an entity at the same time as each other. It is not a "serious" failure (I want the system to continue working as it was before), but I want to tell clients that their pin attempt failed, if the reason for the failure was that "the entity was already pinned".

I don't want to mix "`create_entity_state!` failed because of some unexpected issue" with "`create_entity_state!` failed because another client already pinned this entity", so I need to return better information from `create_entity_state!` to get rid of this TODO. In the "unexpected issue" case, I agree that we can let it fail.

Reply:
Done